drivers/hmis/app/graphql/mutations/split_household.rb

```diff
@@ -0,0 +1,69 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class SplitHousehold < BaseMutation
+    argument :splitting_enrollment_inputs, [Types::HmisSchema::EnrollmentRelationshipInput], required: true
+
+    field :new_household, Types::HmisSchema::Household, null: false
+    field :remaining_household, Types::HmisSchema::Household, null: false
+
+    def resolve(splitting_enrollment_inputs:)
+      map_enrollment_id_to_relationship = splitting_enrollment_inputs.map { |e| [e.enrollment_id, e.relationship_to_hoh] }.to_h
+
+      splitting_enrollment_ids = map_enrollment_id_to_relationship.keys
+      splitting_enrollments = Hmis::Hud::Enrollment.
+        where(id: splitting_enrollment_ids).
+        viewable_by(current_user).
+        includes(:household, :project)
+      access_denied! unless splitting_enrollments.size == splitting_enrollment_inputs.size
+
+      project = splitting_enrollments.map(&:project).uniq.sole
+      access_denied! unless current_permission?(permission: :can_split_households, entity: project)
+
+      donor_household = splitting_enrollments.map(&:household).uniq.sole
+      remaining_enrollments = donor_household.enrollments.where.not(id: splitting_enrollment_ids)
+
+      raise 'Splitting all clients to a new household is invalid' if remaining_enrollments.empty?
+      raise 'This operation would leave behind a household with no HoH, which is not allowed' unless remaining_enrollments.exists?(relationship_to_hoh: 1)
+
+      donor_before_state = donor_household.snapshot_household_state
+      new_household_id = Hmis::Hud::Base.generate_uuid
+
+      Hmis::Hud::Enrollment.transaction do
+        splitting_enrollments.each do |enrollment|
+          enrollment.household_id = new_household_id
+          enrollment.relationship_to_hoh = map_enrollment_id_to_relationship[enrollment.id.to_s]
+
+          enrollment.active_unit_occupancy&.assign_attributes(occupancy_period_attributes: { end_date: Date.current })
+
+          enrollment.save!
+        end
+
+        donor_household.reload
+        donor_after_state = donor_household.snapshot_household_state
+
+        event = Hmis::HouseholdEvent.new
+        event.user = current_user
+        event.household = donor_household
+        event.event_type = Hmis::HouseholdEvent::SPLIT
+        event.event_details = {
+          'receiving_household_id': new_household_id,
+          'before': donor_before_state,
+          'after': donor_after_state,
+        }
+        event.save!
```

Comment:
It might be nice to have this logic contained in the `HouseholdEvent` class, so we can have a consistent place whether the event detail shapes are defined. Right now this json shape is duplicated in both mutations, so the shapes could get out of sync.

```ruby

def self.new_split_event(user:, household:, receiving_household_id:, before_state:, after_state:)
   new(...)
end

def self.new_join_event(user:, household:, donor_household_id:, before_state:, after_state:)
   new(user: user, household: household, ...)
end
```